src/hats/pixel_math/healpix_shim.py

```diff
@@ -134,9 +134,9 @@ def avgsize2order(avg_size_arcmin: np.ndarray) -> np.ndarray:
     np.ndarray of int
         The largest healpix order for which the average size is larger than avg_size_arcmin
     """
-    # resolution = sqrt(4pi / (12 * 4**order)) => order = log2(sqrt(4pi / 12) / resolution)
+    avg_size_arcmin = np.asarray(avg_size_arcmin)
     order_float = np.log2(np.sqrt(np.pi / 3) / np.radians(avg_size_arcmin / 60.0))
-    return np.array(order_float, dtype=int)
+    return order_float.astype(np.int64)
```

Comment:
```suggestion
    return np.clip(order_float.astype(np.int64), a_min=0, a_max=29)
```

I was trying to understand why there were still some unit test failures, and I think that this line is the cause of some frustration.